### PR TITLE
[log4cxx] Add 1.3.0

### PR DESCRIPTION
**Pull request by swebb2066**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/log4cxx/portfile.cmake [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+  URLS https://github.com/apache/logging-log4cxx/commit/4642a50c70b6cbd9b68d7e8dace9c049c8198b07.patch?full_index=1
+  SHA512 4b3628d98d233a2e68d1183a8bb2156c2f1e6f80ab50cfe75a6df799d14bc3c7ba028fbb7ff524c56530a2260be37fd9f3d089422027987b5c8c36e9978c254c
+  FILENAME Make_pkg_config_support_opt_in-4642a50c70b6cbd9b68d7e8dace9c049c8198b07.patch
 )
 
 vcpkg_extract_source_archive(
     SOURCE_PATH ARCHIVE "${ARCHIVE}"
     PATCHES
         fix-find-package.patch
+        ${MAKE_PKG_CONFIG_SUPPORT_OPT_IN}
```
Comment (dg0yt) [inline ports/log4cxx/portfile.cmake:17]: I would expect the port switch it on unconditionally and to fix it as needed.

FTR `vcpkg_fixup_pkgconfig` is guarded by a positive-list condition. This is wrong.

Comment (swebb2066) [reply ports/log4cxx/portfile.cmake:17]: > I would expect the port switch it on unconditionally and to fix it as needed

This was the error when pkf_config support was on:
```
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:134 (message):
  D:/downloads/tools/msys2/21caed2f81ec917b/mingw64/bin/pkg-config.exe
  --exists liblog4cxx failed with error code: 1

      ENV{PKG_CONFIG_PATH}: "D:/p/log4cxx_x64-windows-static/lib/pkgconfig;D:/p/log4cxx_x64-windows-static/share/pkgconfig;D:/installed/x64-windows-static/lib/pkgconfig;D:/installed/x64-windows-static/share/pkgconfig"
      output: Package apr-1 was not found in the pkg-config search path.

  Perhaps you should add the directory containing `apr-1.pc'

  to the PKG_CONFIG_PATH environment variable

  Package 'apr-1', required by 'liblog4cxx', not found
```

How would you fix that from the log4cxx port?

Comment (dg0yt) [reply ports/log4cxx/portfile.cmake:17]: This can't be fix in this port indeed. Poor pkg-config support in `apr`, unfortunately.

**Review by JonLiu1993 — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Pull request merged**